### PR TITLE
Added draw call errors for missing targets

### LINKED ISSUE #525: Detect when the depth/stencil test is used with a missing attachment
We should return an error for these calls.

### PR DESCRIPTION
**Pull request by kvark**
Closes #525 

Note: we also check for missing colors

### CODE REVIEW

File: src/gfx/lib.rs [hunk 1]
```diff
@@ -89,20 +89,11 @@ impl<D: device::Device> Graphics<D> {
...
-    /// Draw a ref batch with instancing
-    pub fn draw_instanced<'a, T: shade::ShaderParam>(&'a mut self,
-                          batch: &'a batch::RefBatch<T>,
-                          data: &'a T,
-                          count: u32,
-                          vertex_offset: u32,
-                          frame: &Frame) {
-        self.renderer.draw_instanced((batch, data, &self.context), count, vertex_offset, frame)
-    }
-
```
Comment (bjz) [inline src/gfx/lib.rs]: Why was this removed?

Comment (kvark) [inline src/gfx/lib.rs]: It's in the `Graphics` struct, which is a simple wrapper around `Renderer` and `Device` for single-threaded mode. It's only a convenience shortcut for simple applications. Once you want anything more advanced, you get rid of `Graphics` or just use `Renderer` directly (it's a public field). Sounds good?

Comment (bjz) [inline src/gfx/lib.rs]: Sounds good. I would really like to figure out a better way to do this...

Comment (bjz) [inline src/gfx/lib.rs]: Ie. not need the `Graphics` thing.

**Comment by bjz** [comment]
Re-running travis

**Comment by bjz** [comment]
Broken on closure syntax
